SETUP.md

```diff
@@ -106,43 +107,143 @@ Then install the Android 28 platform:
 
 ```bash
-sdkmanager "system-images;android-28;google_apis;x86"
-avdmanager create avd --force --name Nexus_5X_API_28 --device "Nexus 5X" -k "system-images;android-28;google_apis;x86" --abi "google_apis/x86"
+brew cask install genymotion
 ```
 
-Execute the following and add it to your `~/.bash_profile`:
+Under OSX High Sierra and later, you'll get a message that you need to
+[approve it in System Preferences > Security & Privacy > General][approve kernel extension].
+
+Do that, and then repeat the line above.
+
+Then make sure the ADB path is set correctly in Genymotion — set
+`Preferences > ADB > Use custom Android SDK tools` to
+`/usr/local/share/android-sdk` (same as `$ANDROID_HOME`)
+
+
+### Linux
+
+#### Install Node, Yarn and friends
+
+We use Yarn to build all of the [celo-monorepo] repo.
+
+Install `nvm` (allows you to manage multiple versions of Node), Node 8 and `yarn`:
 
 ```bash
-export PATH=$ANDROID_HOME/emulator:$ANDROID_HOME/tools:$PATH
+# Installing Node
+sudo apt-get update
+sudo apt-get install nodejs
+
+# Installing Nvm
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
+source ~/.bashrc
+
+# Setting up the right version of Nvm
+nvm install 8
+nvm alias default 8
+
+# Installing Yarn - https://yarnpkg.com/en/docs/install#debian-stable
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+sudo apt-get update && sudo apt-get install yarn
 ```
 
-Run the emulator with:
+#### Installing OpenJDK 8
+
+```
+sudo apt install openjdk-8-jdk
+```
+
+#### Install Android Dev Tools
+
+You can download the complete Android Studio and SDK from the [Android Developer download site](https://developer.android.com/studio/#downloads).
+
+The steps are:
+
+1. Unpack the .zip file you downloaded to an appropriate location for your applications, such as within `/usr/local/` for your user profile, or `/opt/` for shared users.
+   
+   If you're using a 64-bit version of Linux, make sure you first install the [required libraries for 64-bit machines](https://developer.android.com/studio/install#64bit-libs).
+
+1. To launch Android Studio, open a terminal, navigate to the `android-studio/bin/` directory, and execute `studio.sh`.
+
+1. Select whether you want to import previous Android Studio settings or not, then click OK.
+
+1. The Android Studio Setup Wizard guides you through the rest of the setup, which includes downloading Android SDK components that are required for development.
+
+You can find the complete instructions about how to install the tools in Linux environments in the [Documentation page](https://developer.android.com/studio/install#linux).
+
+
+#### Genymotion
+
+Alternatively, you can install Genymotion, a nice emulator manager. It doesn't work with the e2e tests but could help during development. 
```

Comment:
Alternatively to what? Maybe this section was meant to be at the bottom next to the other emulator related instructions?